### PR TITLE
[14.0][FIX+IMP] base_location: Remove zip_id field to _address_fields() function

### PR DESCRIPTION
**Pull request by victoralmau**
Improve tests according to https://github.com/OCA/partner-contact/pull/1180.
Since https://github.com/odoo/odoo/pull/81239 it is not necessary to add the `zip_id` field to `_address_fields()`

Similar in v13: https://github.com/OCA/partner-contact/pull/1194

Related to https://github.com/OCA/partner-contact/pull/1181, now it is necessary to remove the change because odoo has included it in `base_address_city` addon.

Locked by:
- [x] `base_address_city` https://github.com/odoo/odoo/pull/81239

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT33047

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why not doing it in v13 that is already merged?

**Comment by victoralmau** [comment]
> Why not doing it in v13 that is already merged?

Sorry, in v13 it is already created: https://github.com/OCA/partner-contact/pull/1194

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Yes, my fault, sorry

**Review by joao-p-marques — APPROVED** [review]
LGTM :+1: 

Although we should wait for the merge of the Odoo PR locking this before merging this, right?

**Review by chienandalu — APPROVED** [review]

**Comment by victoralmau** [comment]
Odoo has merged the changes, we can then merge this when we apply related commit to OCB, right?

**Pull request merged**
